### ISSUE TITLE
[Enhancement]: Include terraform support to create Amazon OpenSearch Service-managed VPC endpoint

### ISSUE THREAD

**Issue by luvsaxena1** [hashicorp/terraform-provider-aws#27978]
### Description

This is the request for terraform to support creation of Amazon OpenSearch Service-managed VPC endpoint [(link)](https://docs.aws.amazon.com/cli/latest/reference/es/create-vpc-endpoint.html) 
and support access to an Amazon OpenSearch Service domain through the use of an interface VPC endpoint.
 ([linkj](https://docs.aws.amazon.com/cli/latest/reference/es/authorize-vpc-endpoint-access.html))

### Affected Resource(s) and/or Data Source(s)

aws_elasticsearch_domain
aws_elasticsearch_domain_saml_options
aws_elasticsearch_domain_policy

### Potential Terraform Configuration

```terraform
resource "aws_elasticsearch_domain" "example" {
  domain_name           = "example"
  elasticsearch_version = "7.10"

  authorize_vpc_endpoint_access {
    domain_name           = "example"
    "AuthorizedPrincipal" = [ 
     "account_id1",
     "account_id2"
   }
  }

resource "aws_elasticsearch_vpc_endpoint" "vpc-endpoint" {
   "DomainArn" = "string",
   vpc_options {
    subnet_ids = [
      "subnet1",
      "subnet2"
    ]
    security_group_ids = [security_group_id]
  }
   }
}
```

### References

- https://docs.aws.amazon.com/sdk-for-go/api/service/opensearchservice/#OpenSearchService.CreateVpcEndpoint
- https://docs.aws.amazon.com/opensearch-service/latest/APIReference/API_CreateVpcEndpoint.html

### Would you like to implement a fix?

No

**Comment by TheJacobWalters** [comment]
I am interested in working on this.

**Comment by praveenkumarpk** [comment]
We waiting for this...

**Comment by benclapp** [comment]
I wonder if this enhancement should also be included for the paired/similar `aws_opensearch_domain` type resources?

**Comment by tomfankhaenel** [comment]
It would be very good to have this feature. Currently this part is handled by a null_resource and very buggy.

**Comment by raffraffraff** [comment]
@tomfankhaenel would you be happy to share your `null_resource` solution? I'm working on this right now in a multi-account setup. I can authorize other AWS accounts to access OpenSearch, and I can create OpenSearch VPC endpoints multiple AWS accounts. I haven't worked out the destroy operation yet, but I hope to get that fixed this week, if I get time. I'm happy to share my hack once I get that done. I think it might be worthwhile producing some type of hacky solution because, to be perfectly honest, it doesn't look like this is going to make it into the AWS Provider any time soon.

**Comment by mattburgess** [comment]
Looks like this was fixed back in v5.13.0. PR #32435 was marked as resolving the duplicate of this ticket, #29912, but didn't mention this one, so this one's been left as unresolved.

**Comment by mikekinlock** [comment]
I should have some spare time and will work on this task.
